### PR TITLE
Added new simplifier - ExponentiationPolynomial

### PR DESCRIPTION
**Pull request by plugatarev**

### CODE REVIEW

**Comment by artfly** [comment]
Can you please rebase your branch to master? Here's an instruction: https://dev.to/matks/what-it-means-to-rebase-a-pull-request-submitted-on-github-5717
I believe I cannot do it by myself since it requires force push to your fork

**Comment by plugatarev** [comment]
Rebase branch to master

File: arith-simplifier/src/main/java/com/github/artfly/simplifier/simplifier/ExponentiationPolynomial.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    private int factorial(int n) {
+        int result = 1;
+        for (int i = 2; i <= n; i++)
+            result *= i;
+        return result;
+    }
+
+    int getBinomialCoefficient(int k, int n){
```
Comment (artfly) [inline arith-simplifier/src/main/java/com/github/artfly/simplifier/simplifier/ExponentiationPolynomial.java:18]: can simplify to ((n - k + 1) * (n - k + 2) ... * n) / factorial(n - k)

File: arith-simplifier/src/main/java/com/github/artfly/simplifier/simplifier/ExponentiationPolynomial.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            case Expr.Number n -> n;
+            case Expr.Var v -> v;
+            case Expr.Negated n -> new Expr.Negated(exponentiation(n.expr()));
+            case Expr.Binary b -> {
+                if (b.op() != Operator.POWER){
+                    Expr lhs = exponentiation(b.lhs());
+                    Expr rhs = exponentiation(b.rhs());
+                    yield new Expr.Binary(lhs, rhs, b.op());
+                }
+                if (b.lhs() instanceof Expr.Binary bin){
```
Comment (artfly) [inline arith-simplifier/src/main/java/com/github/artfly/simplifier/simplifier/ExponentiationPolynomial.java:33]: please rename to lhsBin

**Review by artfly — CHANGES_REQUESTED** [review]

File: arith-simplifier/src/main/java/com/github/artfly/simplifier/simplifier/ExponentiationPolynomial.java (hunk 1 above)
Comment (plugatarev) [reply arith-simplifier/src/main/java/com/github/artfly/simplifier/simplifier/ExponentiationPolynomial.java:26]: made new method truncatedFactorial by which I count binomial coefficient

**Comment by plugatarev** [comment]
corrected remarks

**Comment by artfly** [comment]
merged, thanks!

**Pull request merged**